core/domain/state_domain_test.py

```diff
@@ -1329,14 +1341,19 @@ def test_cannot_update_interaction_solution_with_non_dict_solution(self):
         }
 
         exploration.init_state.update_interaction_hints(hints_list)
-        exploration.init_state.update_interaction_solution(solution)
+        exploration.init_state.update_interaction_solution(
+            state_domain.Solution.from_dict(
+                exploration.init_state.interaction.id, solution))
 
         self.assertEqual(
             exploration.init_state.interaction.solution.to_dict(), solution)
 
         with self.assertRaisesRegexp(
-            Exception, 'Expected solution to be a dict'):
-            exploration.init_state.update_interaction_solution([])
+            Exception, 'list indices must be integers, not unicode'):
+            exploration.init_state.update_interaction_solution(
```

Comment:
Ah no, to get the error, pass in something like  exploration.init_state.update_interaction_solution('some string'). Basically, pass in something that is not a Solution object, so that error is triggerred. In what you're doing now, an error in from_dict is triggered.